app/grandchallenge/challenges/models.py

```diff
@@ -557,10 +558,51 @@ def remove_admin(self, user):
         unfollow(user=user, obj=self.forum, send_action=False)
 
     @property
-    def accepting_submissions(self):
-        return True in (
-            phase.open_for_submissions for phase in self.phase_set.all()
-        )
+    def status(self):
+        phase_status = {phase.status for phase in self.phase_set.all()}
+        if StatusChoices.OPEN in phase_status:
+            status = StatusChoices.OPEN
+        elif {StatusChoices.COMPLETED} == phase_status:
+            status = StatusChoices.COMPLETED
+        elif StatusChoices.OPENING_SOON in phase_status:
+            status = StatusChoices.OPENING_SOON
+        else:
+            status = StatusChoices.CLOSED
+        return status
+
+    @property
+    def status_badge_string(self):
+        if self.status == StatusChoices.OPEN:
+            detail = [
+                phase.submission_status_string
+                for phase in self.phase_set.all()
+                if phase.status == StatusChoices.OPEN
+            ]
+            if len(detail) > 1:
+                # if there are multiple open phases it is unclear which
+                # status to print, so stay vague
+                detail = ["Accepting submissions"]
+        elif self.status == StatusChoices.COMPLETED:
+            detail = ["Challenge completed"]
+        elif self.status == StatusChoices.CLOSED:
+            detail = ["Not accepting submissions"]
+        else:
```

Comment:
This method looks a lot better! Really easy to follow now. Last thing, we should be defensive in case a future programmer extends `StatusChoices`, as this method should handle all cases. To do that, change the `else:` statement to `elif self.status == StatusChoices.OPENING_SOON`, and add a new `else:` clause that `raise NotImplementedError(f"{self.status} not handled")`. Then if the Enum is extended the future programmer knows where else in the code they need to make a change.

Reply:
Ah right, good point. I also added this to the `submission_status_string` property on the `Phase` model.

Reply:
Nice!